dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DataOwnerControllerTest.kt

```diff
@@ -0,0 +1,245 @@
+package org.dataland.e2etests.tests
+
+import org.dataland.datalandbackend.openApiClient.api.DataOwnerControllerApi
+import org.dataland.datalandbackend.openApiClient.infrastructure.ClientError
+import org.dataland.datalandbackend.openApiClient.infrastructure.ClientException
+import org.dataland.datalandbackend.openApiClient.model.CompanyDataOwners
+import org.dataland.datalandbackend.openApiClient.model.EuTaxonomyDataForNonFinancials
+import org.dataland.e2etests.BASE_PATH_TO_DATALAND_BACKEND
+import org.dataland.e2etests.auth.JwtAuthenticationHelper
+import org.dataland.e2etests.auth.TechnicalUser
+import org.dataland.e2etests.utils.ApiAccessor
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import java.util.UUID
+
+class DataOwnerControllerTest {
+    private val apiAccessor = ApiAccessor()
+    val jwtHelper = JwtAuthenticationHelper()
+    private val dataOwnerApi = DataOwnerControllerApi(BASE_PATH_TO_DATALAND_BACKEND)
+
+    private fun validateDataOwnersForCompany(
+        companyId: UUID,
+        userIds: List<UUID>,
+        dataOwnersForCompany: CompanyDataOwners,
+    ) {
+        val dataOwners = dataOwnersForCompany.dataOwners
+        assertEquals(companyId.toString(), dataOwnersForCompany.companyId)
+        assertEquals(userIds.size, dataOwners.size)
+        userIds.map { it.toString() }.forEach { assertTrue(dataOwners.contains(it)) }
+    }
+
+    @Test
+    fun `test functionality of the data owner`() {
+        val firstCompanyId = UUID.fromString(
+            apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId,
+        )
+        val secondCompanyId = UUID.fromString(
+            apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId,
+        )
+
+        val dataReaderUserId = UUID.fromString("18b67ecc-1176-4506-8414-1e81661017ca")
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+        val frameworkSampleData = apiAccessor.testDataProviderForEuTaxonomyDataForNonFinancials
+            .getTData(1)[0]
+
+        assertFailingApiUploadToCompany(firstCompanyId, frameworkSampleData)
```

Comment:
Why do we need two companies?

Reply:
the idea behind that is that i explicitly test, that after the data owner is added for one company, the operation works only for that and no other company

Reply:
ok, lets keep it